applications/ShapeOptimizationApplication/python_scripts/response_functions/total_volume.py

```diff
@@ -0,0 +1,68 @@
+"""
+This module contains an interface to the available response functions
+"""
+import time as timer
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.ShapeOptimizationApplication as KSO
+from KratosMultiphysics.response_functions.response_function_interface import ResponseFunctionInterface
+
+class TotalVolume(ResponseFunctionInterface):
+    def __init__(self, identifier, response_settings, model):
+        self.identifier = identifier
+        self.response_settings = response_settings
+
+        default_parameters = Kratos.Parameters("""
+        {
+            "response_type"            : "total_volume",
+            "model_part_name"          : "PLEASE_SPECIFY_MODEL_PART_NAME",
+            "output_variable_name"     : "PLEASE_PROVIDE_A_3D_VARIABLE_NAME"
+        }
+        """)
+
+        self.response_settings.ValidateAndAssignDefaults(default_parameters)
+        self.model = model
+
+        # this variable is used to store shape sensitivities of total volume w.r.t. shape
```

Comment:
This seems hacky and is confusing in the `GetNodalGradient` function

We are using the `SHAPE_SENSITIVITY` in other responses as well, since in optimization it we are computing and directly after "getting" the values. 

Did you see an issue with values beeing overriden?
If that is the case we need to create one variable per response/variable pair. 

If not, i would keep the behavior of the other responses.

Reply:
done :)